lib/mhv_logging/client.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+require 'faraday'
+require 'common/client/errors'
+require 'common/client/middleware/response/json_parser'
+require 'common/client/middleware/response/raise_error'
+require 'common/client/middleware/response/snakecase'
+require 'mhv_logging/configuration'
+require 'mhv_logging/client_session'
+require 'mhv_logging/api/audits'
+require 'mhv_logging/api/sessions'
+
+module MHVLogging
+  # Core class responsible for api interface operations
+  class Client
+    include MHVLogging::API::Audits
+    include MHVLogging::API::Sessions
+
+    REQUEST_TYPES = %i(get post).freeze
+    USER_AGENT = 'Vets.gov Agent'
+    BASE_REQUEST_HEADERS = {
+      'Accept' => 'application/json',
+      'Content-Type' => 'application/json',
+      'User-Agent' => USER_AGENT
+    }.freeze
+
+    attr_reader :config, :session
+
+    def initialize(session:)
+      @config = MHVLogging::Configuration.instance
+      @session = MHVLogging::ClientSession.find_or_build(session)
```

Comment:
honestly, I would like to change the DSL for these initialize altogether, but that will happen in a separate refactor.

I want to make the clients such that they are invoked in the following manner.

MHVLogging::ClientSession.user(user_id) -- this sets up a new session given the user if.
MHVLogging::ClientSession.existing_session(session) -- this sets up an existing session

Both of these would return an instance of ClientSession, and calling .new would require you to pass one or the other.